test/api/liskApi.js

```diff
@@ -573,4 +590,110 @@ describe('Lisk.api()', function () {
 			});
 		});
 	});
+
+	describe('#checkReDial', function() {
+
+		it('should check if all the peers are already banned', function () {
+
+			(lisk.api().checkReDial()).should.be.equal(true);
+
+		});
+
+		it('should be able to get a new node when current one is not reachable', function (done) {
+			lisk.api({ node: '123' }).sendRequest('blocks/getHeight', {}, function(result) {
+				(result).should.be.type('object');
+				done();
+			});
+		});
+
+		it('should recognize that now all the peers are banned for mainnet', function () {
+
+			var thisLSK = lisk.api();
+			thisLSK.bannedPeers = lisk.api().defaultPeers;
+
+			(thisLSK.checkReDial()).should.be.equal(false);
+		});
+
+		it('should recognize that now all the peers are banned for testnet', function () {
+
+			var thisLSK = lisk.api({ testnet: true });
+			thisLSK.bannedPeers = lisk.api().defaultTestnetPeers;
+
+			(thisLSK.checkReDial()).should.be.equal(false);
+		});
+
+		it('should recognize that now all the peers are banned for ssl', function () {
+
+			var thisLSK = lisk.api({ssl: true});
+			thisLSK.bannedPeers = lisk.api().defaultSSLPeers;
+
+			(thisLSK.checkReDial()).should.be.equal(false);
+		});
+
+		it('should stop redial when all the peers are banend already', function (done) {
+
+			var thisLSK = lisk.api();
+			thisLSK.bannedPeers = lisk.api().defaultPeers;
+			thisLSK.currentPeer = '';
+
+			thisLSK.sendRequest('blocks/getHeight').then(function (e) {
+				(e.message).should.be.equal('could not create http request to any of the given peers');
+				done();
+			});
+
+		});
+
+	});
+
+	describe('#sendRequest with promise', function () {
+
+		it('should be able to use sendRequest as a promise for GET', function (done) {
+			lisk.api().sendRequest('blocks/getHeight', {}).then(function (result) {
+				(result).should.be.type('object');
+				(result.success).should.be.equal(true);
+				(result.height).should.be.type('number');
+				done();
+			});
+		});
+
+		it('should route the request accordingly if POST can be omited', function (done) {
```

Comment:
This is especially for the API request "accounts/open" - where you want to get privateKey, publicKey and other parameters. I can create them cryptographically without the need to submit the API call to the core. So instead of using the POST request for "accounts/open" I am making it a GET request "accounts?address=address", retreive the values and insert the missing values that would have been retreived for "accounts/open"